File: practical/Final Exam prepare/C/Q2.c
// Author : Akide Liu 
// Date : 4/6/20
// License and copyright notice: GNU General Public License v3.0 
// https://github.com/AkideLiu/C/blob/master/LICENSE
// Description : 

//

#include <stdio.h>
#include <stdlib.h>

void counter(const char *array){
    int count_g = 0;
    int count_a = 0;
    int count_c = 0;
    int count_t = 0;

    for (int i = 0; array[i] != '\0'; ++i) {
        switch (array[i]) {
            case 'G':
                count_g++;
                break;
            case 'A':
                count_a++;
                break;
            case 'C':
                count_c++;
                break;
            case 'T':
                count_t++;
                break;
            default:
                printf("%c , error", array[i]);
        }
    }

    printf("\nThe sequence contains:\n"
           "%d guanine\n"
           "%d adenine\n"
           "%d cytosine\n"
           "%d thymine\n",
           count_g, count_a, count_c, count_t);

}

int main(){
    char *array = (char *) malloc(sizeof(char) * 10);

    printf("Enter the sequence: ");
    scanf("%s", array);

    scanf("%*[^\n]"); scanf("%*c");

    printf("\nSequences: \n%s \n", array);

    counter(array);

    free(array);

    return 0;
}

File: practical/Final Exam prepare/Matlab/Q2.m
clc;
clear;
close all;

% Author : Akide Liu 
% Date : 04 / 06 / 2020 
% License and copyright notice: GNU General Public License v3.0 
% https://github.com/AkideLiu/Matlab/blob/master/LICENSE

% Description : This script ...

sequences = input('Enter the sequence: ','s');

fprintf('Sequences: %s \n', sequences);

count_g = strfind(sequences, 'G');
count_a = strfind(sequences, 'A');
count_c = strfind(sequences, 'C');
count_t = strfind(sequences, 'T');

fprintf('The sequence contains: \n');
fprintf("%d guanine \n", length(count_g));
fprintf("%d adenine \n", length(count_a));
fprintf("%d cytosine \n", length(count_c));
fprintf("%d thymine \n", length(count_t));

